### PR TITLE
Fr 01 (Update script to work with PXC 8.4)

### PR DESCRIPTION
**Pull request by Tusamarco**
The following is to update the script to support PXC 8.4 and above.

### CODE REVIEW

**Comment by Tusamarco** [comment]
@y-trudeau can you spend 10' to check the review to make it work on 8.0 and above.
I have also add the option to define the --defaults-file location and information about the current Source

File: replication_manager5.sh [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+if [ "${DEBUG_LOG}" -a -w "${DEBUG_LOG}" -a ! -L "${DEBUG_LOG}" ]; then
+  exec 9>>"$DEBUG_LOG"
+  exec 2>&9
+  date >&9
+  set -x
+fi
+
+if [ -f /tmp/replication_manager.off ]; then
+    echo "/tmp/replication_manager.off exists, exiting now"
+    exit
```
Comment (EvgeniyPatlan) [inline replication_manager5.sh]:
```suggestion
    exit 1
```

File: replication_manager5.sh [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+
+# set it to the cluster size if you want to distribute the slave, 0 otherwise
+DISTRIBUTE_SLAVE=3 
+
+MYSQL="`which mysql` --connect_timeout=10 -B"
+
+# local override
+if [ -f "/usr/local/etc/replication_manager.cnf" ]; then
+    source /usr/local/etc/replication_manager.cnf
+fi
```
Comment (EvgeniyPatlan) [inline replication_manager5.sh]: if  /usr/local/etc/replication_manager.cnf will be a link to a malicious file the script will load it. So I would suggest to add additional check that it is not a link

File: replication_manager5.sh [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+            fi
+        fi
+    done
+else
+    # cluster node is not sane for this node
+
+    if [ "a$Master_Host" != "a" ]; then 
+        # This node is currently a slave but not in the primary group, bailing out
+        bail_out No
+        send_email "Node $wsrep_node_name is not longer part of the cluster $wsrep_cluster_name, stopping replication" "Failed slave"
```
Comment (EvgeniyPatlan) [inline replication_manager5.sh:504]: do we really need to send emails? it will require sendmail be in dependencies....

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: replication_manager5.sh (hunk 1 above)
Comment (Tusamarco) [reply replication_manager5.sh]: adopted

File: replication_manager5.sh (hunk 2 above)
Comment (Tusamarco) [reply replication_manager5.sh]: added a check to be sure we are NOT referring to a link

File: replication_manager5.sh (hunk 3 above)
Comment (Tusamarco) [reply replication_manager5.sh:504]: I have add a parameter to control this by default NO, we will not send it. 
but no dependency is not needed. If we do not have a mailer we will automatically not send it, period.

**Comment by Tusamarco** [comment]
@y-trudeau ... can you please give a quick review ?

**Review by EvgeniyPatlan — APPROVED** [review]

**Comment by y-trudeau** [comment]
All good on my side

**Pull request merged**